scripts/accelerator.ts

```diff
@@ -6,7 +6,7 @@ import { zodSchemaIsAsync } from "./utils/zodSchemaIsAsync";
 
 declare module "zod" {
 	interface ZodType {
-		accelerator?: ZodAcceleratorParser<this>;
+		accelerator?: ZodAcceleratorParser;
```

Comment:
```ts
declare module "zod" {
	interface ZodType<
		Output = any,
		Def extends ZodTypeDef = zod.ZodTypeDef,
		Input = Output,
	> {
		accelerator?: ZodAcceleratorParser<ZodType<Output, Def, Input>>;
	}
}
```

This solution seems the best to me, it allows you to save a typing but not have any problems ts(2589).